src/Utils/ICSCreator.ts

```diff
@@ -146,4 +147,63 @@ export class ICSCreator {
         await ICSCreator.createClubICS(competition.getFetcher(),
             competition.getMatches(), path, title, meta);
     }
+
+    /**
+     * Create ICS files for each official's matches
+     * @param fetcher The fetcher
+     * @param competitions All competitions to include
+     */
+    public static async createOfficialICS(
+        fetcher: Fetcher,
+        competitions: Competition[]
+    ) {
+        const matches = competitions.map(e => e.getMatches()).flat();
+        const officialsMap = new Map<string, {
+            matches: Match[],
+            official: Official
+        }>();
+
+        // Group matches by official
+        for (const match of matches) {
+            for (const official of match.getOfficials()) {
+                const officialKey = `${official.name}${
+                    official.country ? `-${official.country}` : ""
+                }`;
+
+                if (!officialsMap.has(officialKey)) {
+                    officialsMap.set(officialKey, {
+                        matches: [],
+                        official
+                    });
+                }
+
+                officialsMap.get(officialKey).matches.push(match);
+            }
+        }
+
+        // Create calendar file for each official
+        const promises = [];
+        for (const [officialKey, data] of officialsMap) {
+            const path = `officials/${officialKey}/all-matches`;
+            const title = `${data.official.name}${
+                data.official.country ? ` (${data.official.country})` : ""
+            } - All Matches`;
+
+            promises.push(ICS.writeToFile(
+                fetcher,
+                data.matches,
+                title,
+                path,
+                null,
+                {
+                    type: "official",
```

Comment:
I think we need to refactor this to instead of having an array of all paths with a given type (which needs sorting and filtering to get the right ones), to add two separate arrays to the `paths.json`.

A single array of all competition paths, and a single array of all officials with their corresponding ICS.